xds/src/main/java/io/grpc/xds/internal/certprovider/FileWatcherCertificateProvider.java

```diff
@@ -110,17 +81,18 @@ public void start() {
 
   @Override
   public void close() {
-    if (scheduledHandle != null) {
-      scheduledHandle.cancel();
-      scheduledHandle = null;
+    shutdown = true;
```

Comment:
Is `FileWatcherCertificateProvider` thread safe?

Reply:
A CertificateProvider instance is only accessed from the `CertificateProviderStore` (which is a singleton) and the `createOrGetProvider()` method as well as `CertificateProviderStore.Handle.close()` method are synchronized so there are no unsafe multi-threaded accesses in these contexts.

Reply:
`shutdown` and `scheduledFuture` are not synchronized, and they are not accessed and updated atomically , so there could be race.

Reply:
made `close()` and `scheduleNextRefreshCertificate` synchronized